ui/app/templates/rules.html

```diff
@@ -37,57 +43,110 @@
-        <button ng-show="!rule_id" class="btn btn-default btn-xs" ng-click="openDeleteModal(rule)">Delete</button>
-        <button ng-show="!rule_id" class="btn btn-default btn-xs" ng-click="openDuplicateModal(rule)">Duplicate</button>
-        <a ng-show="!rule_id" class="btn btn-default btn-xs" ng-href="/rules/{{ rule.rule_id }}">History</a>
+        <button class="btn btn-xs btn-primary" ng-click="signoff(rule.scheduled_change)"
+                ng-show="!rule_id && show_sc && rule.scheduled_change !== null && ! isEmpty(rule.scheduled_change.required_signoffs) && ! rule.scheduled_change.signoffs.hasOwnProperty(current_user)">
+            Signoff as...
+        </button>
+        <button class="btn btn-xs btn-danger" ng-click="revokeSignoff(rule.scheduled_change)"
+                ng-show="!rule_id && show_sc && rule.scheduled_change !== null && ! isEmpty(rule.scheduled_change.required_signoffs) && rule.scheduled_change.signoffs.hasOwnProperty(current_user)">
+            Revoke your Signoff
+        </button>
+        <button ng-show="!rule_id && show_sc && rule.scheduled_change.change_type === 'insert'" class="btn btn-info btn-xs"
+                ng-click="openEditScheduledRuleChangeModal(rule)" title="{{ humanizeDate(rule.scheduled_change.when) }}">
+            Pending Creation <span ng-bind-html="formatMoment(rule.scheduled_change.when)"></span>
+        </button>
+        <button ng-show="!rule_id && show_sc && rule.scheduled_change.change_type === 'update'" class="btn btn-warning btn-xs"
+                ng-click="openEditScheduledRuleChangeModal(rule)" title="{{ humanizeDate(rule.scheduled_change.when) }}">
+            Pending Update <span ng-bind-html="formatMoment(rule.scheduled_change.when)"></span>
+        </button>
+        <button ng-show="!rule_id && show_sc && rule.scheduled_change.change_type === 'delete'" class="btn btn-danger btn-xs"
+                ng-click="openEditScheduledRuleChangeModal(rule)" title="{{ humanizeDate(rule.scheduled_change.when) }}">
+            Pending Delete <span ng-bind-html="formatMoment(rule.scheduled_change.when)"></span>
+        </button>
+        <button ng-show="!rule_id && rule.scheduled_change === null" class="btn btn-warning btn-xs"
+                ng-click="openNewScheduledRuleChangeModal(rule)">
+            Schedule an Update
+        </button>
+        <button ng-show="!rule_id && rule.scheduled_change === null" class="btn btn-danger btn-xs"
+                ng-click="openNewScheduledDeleteModal(rule)">
+            Schedule for Deletion
+        </button>
+        <button ng-show="!rule_id && rule.scheduled_change !== null && rule.scheduled_change.change_type !== 'insert'"
+                class="btn btn-default btn-xs" ng-click="openUpdateModal(rule)">
+            Update
+        </button>
+        <button ng-show="!rule_id && rule.scheduled_change !== null && rule.scheduled_change.change_type !== 'insert'"
```

Comment:
I don't think we should show the Update/Delete buttons, if a scheduled change is active. If you ask for a deletion while a pending change is active, you end up with this (expected) error:

> "Couldn't perform the request DELETE due a conflict with a scheduled change. Cannot delete rows that have changes scheduled."

Duplicate/History are good to keep, thought, IMO.

Reply:
I think you're right that we should hide the Delete button, but you can still update when there's a change scheduled, so long as you don't try to modify any of the same rows the Change does.

Good point about Duplicate though, that should be shown regardless.

Reply:
It may also be worth noting that once https://bugzilla.mozilla.org/show_bug.cgi?id=1355076 is fixed, the Update/Delete/Duplicate buttons should be hidden for any channel that has required signoffs (since you must use Scheduled Changes). It's still good to make sure works properly for Nightly though.